rust/src/pgsql/pgsql.rs

```diff
@@ -48,23 +48,23 @@ pub enum PgsqlEvent {
 
 #[repr(u8)]
 #[derive(Copy, Clone, PartialOrd, PartialEq, Eq, Debug)]
-pub enum PgsqlTxProgress {
+pub(crate) enum PgsqlTxProgress {
     TxInit = 0,
     TxReceived,
     TxDone,
     TxFlushedOut,
 }
 
 #[derive(Debug)]
-pub struct PgsqlTransaction {
-    pub tx_id: u64,
-    pub tx_req_state: PgsqlTxProgress,
-    pub tx_res_state: PgsqlTxProgress,
-    pub request: Option<PgsqlFEMessage>,
-    pub responses: Vec<PgsqlBEMessage>,
+pub(crate) struct PgsqlTransaction {
+    pub(crate) tx_id: u64,
+    pub(crate) tx_req_state: PgsqlTxProgress,
+    pub(crate) tx_res_state: PgsqlTxProgress,
+    pub(crate) requests: Vec<PgsqlFEMessage>,
+    pub(crate) responses: Vec<PgsqlBEMessage>,
 
-    pub data_row_cnt: u64,
-    pub data_size: u64,
+    pub(crate) data_row_cnt: u64,
+    pub(crate) data_size: u64,
```

Comment:
You can live offer the inner `pub(crate)`.  An inner item can't be any more visible than its outer item.

Reply:
🙇🏽‍♀️